Pagination page changes (#120171)
elastic/elasticsearch: docs/reference/search/search-your-data/paginate-search-results.asciidoc

[[paginate-search-results]]
=== Paginate search results

> Pagination provides a structured way for users to interact with search results by breaking them down into smaller, manageable pieces or "pages." Whether allowing users to select a specific range of pages or implementing an “infinite scroll” experience, pagination helps create a tailored and intuitive user experience.

Reviewer: ```suggestion
Pagination organizes search results into manageable pages, enhancing user interaction. Whether selecting specific page ranges or using infinite scroll, pagination offers a tailored user experience.
```

The three commonly used pagination techniques are:

> * <<from-and-size-pagination, From and size pagination>>: This method is ideal for creating a defined list of pages that users can select to navigate through the results.

Reviewer: ```suggestion
* <<from-and-size-pagination, From and size pagination>>: Ideal for creating a list of pages that users can navigate.
```

> * <<search-after, Search after>>: Designed for seamless navigation, this technique supports “infinite scroll” experiences or enables users to load additional results by pressing a “next” button.

Reviewer: ```suggestion
* <<search-after, Search after>>: Supports infinite scroll or allows loading additional results with a "next" button.
```

> * <<scroll-search-results, Scroll>>: Historically used to retrieve all matching documents for display or export. However, we now recommend using the <<search-after, search after>> method in combination with the <<point-in-time-api, point in time API>> for improved efficiency and reliability.

Reviewer: ```suggestion
* <<scroll-search-results, Scroll>>: Historically used to retrieve all matching documents. The <<search-after, search after>> method with the <<point-in-time-api, point in time API>> is now recommended for better efficiency and reliability.
```

once. Search requests usually span multiple shards. Each shard must load its
requested hits and the hits for any previous pages into memory. For deep pages
or large sets of results, these operations can significantly increase memory and
CPU usage, resulting in degraded performance or node failures if not managed correctly.

By default, you cannot use `from` and `size` to page through more than 10,000
hits. This limit is a safeguard set by the
<<index-max-result-window,`index.max_result_window`>> index setting. If you need
to page through more than 10,000 hits, use the <<search-after,`search_after`>>
parameter instead.

> Pagination is stateless, meaning there is no guarantee that the order of search results will remain consistent when users navigate back and forth between pages. If maintaining consistent result order is desired, the preferred approach is to use the <<point-in-time-api, point in time (PIT) API>> to implement a stateful pagination technique.

Reviewer: ```suggestion
Pagination is stateless, so search result order may change when navigating between pages. To maintain consistent order, use the <<point-in-time-api, point in time API>> for stateful pagination.
```
